examples/experimental/horizontal-pod-autoscaling/README.md

```diff
@@ -0,0 +1,189 @@
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Horizontal Pod Autoscaler
+
+This example presents horizontal autoscaling of pods: an experimental feature delivered in version 1.1 of Kubernetes
+which automatically sets the number of pods in a replication controller (or deployment), base on CPU / memory usage.
+
+
+## Prerequisites
+
+This example requires a running Kubernetes cluster in the version 1.1 with experimental API enabled on API server (```--runtime-config=experimental/v1alpha1=true```),
+and horizontal pod autoscaler controller turned on in controller manager (```--enable-horizontal-pod-autoscaler=true```).
+This can be simply achieved on GCE by exporting ```KUBE_ENABLE_HORIZONTAL_POD_AUTOSCALER=true``` before running ```kube-up.sh``` script.
+
+The required version of kubectl is also 1.1.
+
+
+## Step One: Run & expose php-apache server
+
+To demonstrate horizontal pod autoscaler we will use a custom docker image based on php-apache server.
+The image can be found [here](image/).
+It defines [index.php](image/index.php) page which performs some CPU intensive computations.
+
+First, we will start a replication controller running the image and expose it as an external service:
+
+```console
+$ kubectl run php-apache --image=gcr.io/dev-jsz/php-apache --replicas=1
+replicationcontroller "php-apache" created
+
+$ kubectl expose rc php-apache --port=80 --type=LoadBalancer
+
+```
+
+Now, we will wait some time and verify that both the replication controller and the service were correctly created and are running. We will also determine the IP address of the service:
+
+```console
+$ kubectl get pods
+NAME               READY     STATUS    RESTARTS   AGE
+php-apache-wa3t1   1/1       Running   0          12m
+
+$ kubectl describe services php-apache | grep "LoadBalancer Ingress"
+LoadBalancer Ingress:	146.148.24.244
+
+```
+
+We may now check that php-apache server works correctly by calling ```curl``` with the service's IP:
+
+```console
+$ curl http://146.148.24.244
+OK
+```
+
+## Step Two: Create horizontal pod autoscaler
+
+Now that the server is running, we will create a horizontal pod autoscaler for it.
+To create it, we will use [hpa-php-apache.yaml](hpa-php-apache.yaml) file, which look like this:
+
+```yaml
+apiVersion: experimental/v1alpha1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: php-apache
+  namespace: default
+spec:
+  maxReplicas: 10
+  minReplicas: 1
+  scaleRef:
+    kind: ReplicationController
+    name: php-apache
+    namespace: default
+  target:
+    quantity: 100m
+    resource: cpu
+```
+
+Please note that it sets the target CPU consumption to 100 mili-cores.
+
+We will create the autoscaller by executing the following command:
+
+```console
+$ kubectl create -f ./examples/experimental/horizontal-pod-autoscaling/hpa-php-apache.yaml
+horizontalpodautoscaler "php-apache" created
+```
+
+We may check the current status of autoscaler by running:
+
+```console
+$ kubectl get hpa
```

Comment:
I think this will be changing soon:

Be aware depending on merge order:
https://github.com/kubernetes/kubernetes/pull/15206